frontend/src/components/DrawerDetails/ResourceDetailsDrawer.tsx

```diff
@@ -0,0 +1,152 @@
+                                                                 onClose,
+                                                                 resourceType,
+                                                             }) => {
+    const [resourceDetails, setResourceDetails] = useState<any | null>(null);
+    const [fetching, setFetching] = useState(false);
+
+    const fetchResourceData = async (
+        resourceType: string,
+        resourceName: string,
+        namespace: string
+    ) => {
+        setFetching(true);
+        try {
+            if (resourceType !== "Helm") {
+                const details = await getResource(resourceType, resourceName, namespace);
+                setResourceDetails(details);
+                console.log(details);
+            } else {
+                const details = await fetchRelease(resourceName, namespace);
+                const history = await fetchReleaseHistory(resourceName, namespace);
+
+                const combinedDetails = [details, history];
+
+                setResourceDetails(combinedDetails);
+                console.log(combinedDetails);
+
+            }
+        } catch (error: unknown) {
+            if (error instanceof Error) {
+                console.error("Error fetching resource details:", error.message);
+            } else {
+                console.error("Unknown error:", error);
+            }
+            setResourceDetails(null);
+        }
+        setFetching(false);
+    };
+
+    useEffect(() => {
+        if (visible && record) {
+            if ("namespace" in record && "name" in record) {
+                const namespace = record.namespace as string;
+                const resourceName = record.name as string;
+                fetchResourceData(resourceType, resourceName, namespace);
+            }
+        }
+    }, [visible, record]);
+
+    const renderObject = (obj: any, parentKey = ""): React.ReactNode => {
+        if (!obj || typeof obj !== "object") return null;
+
+        return Object.entries(obj).map(([key, value]) => {
+            const panelKey = parentKey ? `${parentKey}.${key}` : key;
+
+            if (value && typeof value === "object") {
+                return (
+                    <Collapse key={panelKey} className={styles.collapse}>
+                        <Panel header={key} key={panelKey}>
+                            {renderObject(value, panelKey)}
+                        </Panel>
+                    </Collapse>
+                );
+            } else {
+                return (
+                    <div key={panelKey} className={styles.singleDetail}>
+                        <div className={styles.key}>{key}:</div>
+                        <Tooltip title={value?.toString() || "—"}>
+                            <div className={styles.value}>{value?.toString() || "—"}</div>
+                        </Tooltip>
+                    </div>
+                );
+            }
+        });
+    };
```

Comment:
I don’t like how this is displayed—it took me a while to figure out what I was looking at. That said, I’ll admit I don’t have a good solution at the moment.

![image](https://github.com/user-attachments/assets/d80f49c2-14d9-42ba-9cda-cba29cde924f)

Reply:
Ja zaczaiłem od razu xd